client/admin/client.go

```diff
@@ -226,7 +227,7 @@ func (c *clientImpl) createContext(parent context.Context) (context.Context, con
 	if parent == nil {
 		return context.WithTimeout(context.Background(), c.timeout)
 	}
-	return context.WithTimeout(parent, c.timeout)
+	return context.WithTimeout(client.PropagateHeaders(parent), c.timeout)
```

Comment:
Context is not propagated between calls by default. This call copies incoming headers to outgoing headers.